utils.sh

```diff
@@ -2,6 +2,22 @@
 
 set -o pipefail
 
+function retry_with_timeout() {
+  retries=$1
+  timeout_duration=$2
+  command=${@:3}
+
+  for i in $(seq $retries); do
```

Comment:
i is never used and retries should be quoted, being a function parameter.

```suggestion
  for _ in $(seq "$retries"); do
```